mne/preprocessing/xdawn.py

```diff
@@ -129,31 +126,139 @@ def least_square_evoked(epochs, return_toeplitz=False):
         If return_toeplitz is true, return the toeplitz matrix for each event
         type in epochs.event_id.
     """
-    if not isinstance(epochs, _BaseEpochs):
-        raise ValueError('epochs must be an instance of `mne.Epochs`')
-
-    events = epochs.events.copy()
-    events[:, 0] -= events[0, 0] + int(epochs.tmin * epochs.info['sfreq'])
-    data = _construct_signal_from_epochs(epochs)
-    evoked_data, toeplitz = _least_square_evoked(data, events, epochs.event_id,
-                                                 tmin=epochs.tmin,
-                                                 tmax=epochs.tmax,
-                                                 sfreq=epochs.info['sfreq'])
+
+    events = events.copy()
+    events[:, 0] -= events[0, 0] + int(tmin * sfreq)
+    data = _construct_signal_from_epochs(events, sfreq, tmin, tmax,
+                                         epochs_data)
+    evoked_data, toeplitz = _least_square_evoked(data, events, event_id,
+                                                 tmin=tmin, tmax=tmax,
+                                                 sfreq=sfreq)
     evokeds = dict()
-    info = cp.deepcopy(epochs.info)
     for name, data in evoked_data.items():
-        n_events = len(events[events[:, 2] == epochs.event_id[name]])
-        evoked = EvokedArray(data, info, tmin=epochs.tmin,
-                             comment=name, nave=n_events)
-        evokeds[name] = evoked
+        evokeds[name] = data
 
     if return_toeplitz:
         return evokeds, toeplitz
 
     return evokeds
 
 
-class Xdawn(TransformerMixin, ContainsMixin):
+class _Xdawn(object):
+    """Private class containing common functions for Xdawn and
+       XdawnTransformer.
+    """
+    def __init__(self, n_components, signal_cov, reg):
+        self.n_components = n_components
+        self.signal_cov = signal_cov
+        self.reg = reg
+
+    def _get_signal_cov(self, epochs_data):
```

Comment:
this function is rather setting than getting ...
maybe call it `_set_signal_cov`